### ISSUE TITLE
Legend label not showing for BarChart if BarEntries only have ONE y value supplied 

### ISSUE THREAD

**Issue by oseparovic** [PhilJay/MPAndroidChart#4204]
**Summary**
Legend label in stacked bar chart not rendering next to legend color if BarEntry contains only ONE y value e.g.

```
BarEntry entry = new BarEntry(i, new float[]{volumeForEntry});
```

or 

```
BarEntry entry = new BarEntry(i, volumeForEntry);
```

**Expected Behavior**
When only one y value is specified for all bar entries it should still show the name of the legend next to the one colour

**Possible Solution**
Likely this is a conditional case like i <= 0 behind the logic for whether to render legend text. Same issue does not exist for the colours only the label

**Device (please complete the following information):**
OnePlus 3
8.0
3.0.3

**Additional Context**
Example (should say "Volume" next to legend): https://imgur.com/a/U17pZMq 

**Workaround**
Even if you only supply one stack label to `dataset.setStackLabels` and one color to `dataset.setColors` you can still have your legend text. Just spoof the BarEntry to always have more than one y point e.g.

```
new BarEntry(i, new float[]{volumeForEntry, 0});
```

instead of:

```
new BarEntry(i, new float[]{volumeForEntry});
```

**Comment by almic** [comment]
I'm confused, you say that all bars have the same y-value, and that causes the label to only partially render. However in the image, the bars clearly have different y-values.

Either way, can you please create a gist with the code where you call `new BarDataSet()` and `BarDataSet.setStackLabels()`? Technically if you create a stacked data set with only one stack, then the `LegendRenderer` actually uses the label set initially when you do `new BarDataSet()`.

In other words, a one stacked bar chart is no different from a non-stacked bar chart, so calling `setStackLabels()` will do literally nothing for one stack bar charts.

**Comment by almic** [comment]
Honestly not sure what to do about this, as there is no easy way to change the way a stacked bar chart is considered stacked without adding a lot of clutter code. All I can tell you is that you cannot use stacked labels on a one stack chart because the render won't use that as the chart isn't stacked anyway.

**Comment by oseparovic** [comment]
It's probably just an issue of me using the wrong functionality for the problem then. I was initially going to add more than one stack to each bar but reverted back to just the one data point per BarEntry and discovered the problem.

If it's not feasible to make stack labels function for a one stack bar chart then just close this issue and it can remain here as a reference of what to do instead.

**Comment by almic** [comment]
👍

**Comment by folab33** [comment]
I ran into the same issue as my dataset is dynamic, the work around I use is to set the dataset label to the legend label if the dataset contains only one entry and empty string if dataset contains more than one entry
